### DISCUSSION TITLE
if_else_lamellar

### DISCUSSION THREAD

**Discussion by kittygg** [neperfepx/neper#639] [Q&A] [answered]
Hi, Dr. Quey
In 2D domain, for the model which grain size obeys the normal distribution, how to layer the grains (a specified grain size range) using the lamellar code? Thanks!
[code.txt](https://github.com/neperfepx/neper/files/10372207/code.txt)

**Comment by rquey** [comment]
I don't understand the question, so, please clarify (by *editing* it) and provide your code in-line, using [markdown](https://docs.github.com/github/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) for proper formatting.  Thanks.

**Reply by kittygg** [reply]
neper -T -n 100  -dim 2   -reg 1  -domain "square(10,10)"  -morpho "gg,diameq:normal(0.6,0.1)"  -statcell diameq  -o 1
awk '{if ($1>0.6) {print "lamellar(w=0.2)"} else {print "?"}}' 1.stcell | nl  > 2
neper -T -n 100::from_morpho  -dim 2  -reg 1 -domain "square(10,10)"  -morpho "gg::msfile(2)"  -o 3

**Reply by kittygg** [reply]
Hi, Dr. Quey
Sorry, I want to generate the structure shown in the following figure. Layered structure is formed in the grains with specified diameq (such as 0.2~0.5 mm). Thanks!
![example](https://user-images.githubusercontent.com/110077741/212604093-9f1df26d-9404-4a06-92ba-49b52eea9274.png)

**Comment by rquey** [comment, marked as answer]
You have to proceed in 2 steps:
1. Generate the 1st-scale tessellation and export the cell `diameq` values.
2. Write a [multiscale cell file](https://neper.info/doc/fileformat.html#multiscale-cell-file) based on the `diameq` to specify the morphology at scale 2 in a second run.

```
neper -T -n 100 -morpho gg -dim 2 -statcell diameq -o 1scale
awk '{if ($1<0.15) print NR,"diameq:1"; else print NR,"lamellar(w=0.05)"}' 1scale.stcell > scale2
neper -T -n 100::from_morpho -morpho "gg::msfile(scale2)" -dim 2 -o 2scale
neper -V 2scale.tess -datacellcol "scaleid(1)" -dataedgerad 0.002 -print 2scale
```
![image](https://user-images.githubusercontent.com/2659376/213790639-14df8de9-6178-428a-b43d-7d6bd567b4c2.png)
